IndexedDbGetAllEntries/explainer.md

```diff
@@ -0,0 +1,196 @@
+# IndexedDB: getAllEntries()
+
+## Author:
+- [Steve Becker](https://github.com/SteveBeckerMSFT)
+
+## Participate
+- https://github.com/w3c/IndexedDB/issues/206
+
+## Introduction
+
+[`IndexedDB`](https://developer.mozilla.org/en-US/docs/Web/API/IndexedDB_API) is a transactional database for client-side storage.  Each record in the database contains a key-value pair.  [`getAll()`](https://developer.mozilla.org/en-US/docs/Web/API/IDBObjectStore/getAll) enumerates database record values sorted by key in ascending order.  [`getAllKeys()`](https://developer.mozilla.org/en-US/docs/Web/API/IDBObjectStore/getAllKeys) enumerates database record primary keys sorted by key in ascending order.
+
+This explainer proposes a new operation, `getAllEntries()`, which combines [`getAllKeys()`](https://developer.mozilla.org/en-US/docs/Web/API/IDBObjectStore/getAllKeys) with [`getAll()`](https://developer.mozilla.org/en-US/docs/Web/API/IDBObjectStore/getAll) to enumerate both primary keys and values at the same time.  For an [`IDBIndex`](https://developer.mozilla.org/en-US/docs/Web/API/IDBIndex), `getAllEntries()` also provides the record's index key in addition to the primary key and value.  Lastly, `getAllEntries()` offers a new option to enumerate records sorted by key in descending order.
+
+## WebIDL
+
+```js
+dictionary IDBGetAllEntriesOptions {
+  // A key or an `IDBKeyRange` identifying the records to retrieve.
+  any query = null;
+
+  //  The maximum number of results to retrieve.
+  [EnforceRange] unsigned long count;
+
+  // Determines how to enumerate and sort results.
+  // Use 'prev' to enumerate and sort results by key in descending order.
+  IDBCursorDirection direction = 'next';
+}; 
+
+[Exposed=(Window,Worker)]
+partial interface IDBObjectStore {
+  // After the `getAllEntries()` request completes, the `IDBRequest::result` property
```

Comment:
We might have to define Exceptions; maybe say it is a same with GetAll, GetAllKeys or any new exception?

Reply:
Yes, if this gets added to the Editor's draft of the W3C IndexedDB spec, then we will need to describe both the algorithm and the error case in much more detail.  For the explainer, we don't have to necessarily go into those details since the function will have the same failures as the existing functions getAll() and getAllKeys().